### ISSUE TITLE
Breaking change introduced in 0.15.10

### ISSUE THREAD

**Issue by BBosman** [systemjs/builder#513]
I'm building a webapp using Aurelia. I use `aurelia-bundler`, which itself uses `systemjs-builder`. 

Last week everything was working, but this morning it wasn't. Bundling gives this error:
```
[08:36:23] Error on instantiate for npm:aurelia-loader-default@1.0.0-beta.1.1.2/aurelia-loader-default.js at file:///Users/BBosman/Development/ExternalApps/Aurelia/jspm_packages/npm/aurelia-loader-default@1.0.0-beta.1.1.2/aurelia-loader-default.js
	Loading npm:aurelia-loader-default@1.0.0-beta.1.1.2.js
	Error: Multiple anonymous defines.
    at ParseTreeTransformer.AMDDependenciesTransformer.transformCallExpression (/Users/BBosman/Development/ExternalApps/Aurelia/node_modules/systemjs-builder/compilers/amd.js:98:13)
    at CallExpression.$__super.transform (/Users/BBosman/Development/ExternalApps/Aurelia/node_modules/traceur/bin/traceur.js:6649:28)
    at ParseTreeTransformer.transformAny (/Users/BBosman/Development/ExternalApps/Aurelia/node_modules/traceur/bin/traceur.js:9193:44)
    at ParseTreeTransformer.transformExpressionStatement (/Users/BBosman/Development/ExternalApps/Aurelia/node_modules/traceur/bin/traceur.js:9504:31)
    at ExpressionStatement.$__super.transform (/Users/BBosman/Development/ExternalApps/Aurelia/node_modules/traceur/bin/traceur.js:7120:28)
    at ParseTreeTransformer.transformAny (/Users/BBosman/Development/ExternalApps/Aurelia/node_modules/traceur/bin/traceur.js:9193:44)
    at ParseTreeTransformer.transformList (/Users/BBosman/Development/ExternalApps/Aurelia/node_modules/traceur/bin/traceur.js:9200:34)
    at ParseTreeTransformer.transformFunctionBody (/Users/BBosman/Development/ExternalApps/Aurelia/node_modules/traceur/bin/traceur.js:9574:31)
    at FunctionBody.$__super.transform (/Users/BBosman/Development/ExternalApps/Aurelia/node_modules/traceur/bin/traceur.js:7293:28)
    at ParseTreeTransformer.transformAny (/Users/BBosman/Development/ExternalApps/Aurelia/node_modules/traceur/bin/traceur.js:9193:44)
    at ParseTreeTransformer.transformFunctionExpression (/Users/BBosman/Development/ExternalApps/Aurelia/node_modules/traceur/bin/traceur.js:9596:25)
    at ParseTreeTransformer.AMDDependenciesTransformer.transformFunctionExpression (/Users/BBosman/Development/ExternalApps/Aurelia/node_modules/systemjs-builder/compilers/amd.js:61:69)
    at FunctionExpression.$__super.transform (/Users/BBosman/Development/ExternalApps/Aurelia/node_modules/traceur/bin/traceur.js:7339:28)
    at ParseTreeTransformer.transformAny (/Users/BBosman/Development/ExternalApps/Aurelia/node_modules/traceur/bin/traceur.js:9193:44)
    at ParseTreeTransformer.transformBinaryExpression (/Users/BBosman/Development/ExternalApps/Aurelia/node_modules/traceur/bin/traceur.js:9299:26)
    at BinaryExpression.$__super.transform (/Users/BBosman/Development/ExternalApps/Aurelia/node_modules/traceur/bin/traceur.js:6557:28)
    at ParseTreeTransformer.transformAny (/Users/BBosman/Development/ExternalApps/Aurelia/node_modules/traceur/bin/traceur.js:9193:44)
    at ParseTreeTransformer.transformExpressionStatement (/Users/BBosman/Development/ExternalApps/Aurelia/node_modules/traceur/bin/traceur.js:9504:31)
    at ExpressionStatement.$__super.transform (/Users/BBosman/Development/ExternalApps/Aurelia/node_modules/traceur/bin/traceur.js:7120:28)
    at ParseTreeTransformer.transformAny (/Users/BBosman/Development/ExternalApps/Aurelia/node_modules/traceur/bin/traceur.js:9193:44)
    at ParseTreeTransformer.transformList (/Users/BBosman/Development/ExternalApps/Aurelia/node_modules/traceur/bin/traceur.js:9200:34)
    at ParseTreeTransformer.transformBlock (/Users/BBosman/Development/ExternalApps/Aurelia/node_modules/traceur/bin/traceur.js:9317:31)
    at Block.$__super.transform (/Users/BBosman/Development/ExternalApps/Aurelia/node_modules/traceur/bin/traceur.js:6612:28)
```

After some searching I figured out that an `npm install` I did this morning updated `systemjs-builder` to 0.15.10. Downgrading to 0.15.9 fixed it.

No idea what the underlying issue is, but patch releases shouldn't include breaking changes. Let me know if you need anything to help this get fixed.

**Comment by druvisc** [comment]
Similar error:
```
[15:37:23] 'jspm' errored after 11 s
[15:37:23] TypeError: Error compiling amd module "github:tildeio/route-recognizer@0.1.9/dist/route-recognizer.js" at jspm_packages/github/tildeio/route-recognizer@0.1.9/dist/route-recognizer.js
    Cannot read property 'parameters' of undefined
    at ParseTreeTransformer.AMDDefineRegisterTransformer.transformCallExpression (/Invend/apps/egbooking/node_modules/systemjs-builder/compilers/amd.js:227:76)
    at CallExpression.$__super.transform (/Invend/apps/egbooking/node_modules/systemjs-builder/node_modules/traceur/bin/traceur.js:6649:28)
    at ParseTreeTransformer.transformAny (/Invend/apps/egbooking/node_modules/systemjs-builder/node_modules/traceur/bin/traceur.js:9193:44)
    at ParseTreeTransformer.transformExpressionStatement (/Invend/apps/egbooking/node_modules/systemjs-builder/node_modules/traceur/bin/traceur.js:9504:31)
    at ExpressionStatement.$__super.transform (/Invend/apps/egbooking/node_modules/systemjs-builder/node_modules/traceur/bin/traceur.js:7120:28)
    at ParseTreeTransformer.transformAny (/Invend/apps/egbooking/node_modules/systemjs-builder/node_modules/traceur/bin/traceur.js:9193:44)
    at ParseTreeTransformer.transformList (/Invend/apps/egbooking/node_modules/systemjs-builder/node_modules/traceur/bin/traceur.js:9200:34)
    at ParseTreeTransformer.transformBlock (/Invend/apps/egbooking/node_modules/systemjs-builder/node_modules/traceur/bin/traceur.js:9317:31)
    at Block.$__super.transform (/Invend/apps/egbooking/node_modules/systemjs-builder/node_modules/traceur/bin/traceur.js:6612:28)
    at ParseTreeTransformer.transformAny (/Invend/apps/egbooking/node_modules/systemjs-builder/node_modules/traceur/bin/traceur.js:9193:44)
    at ParseTreeTransformer.transformToBlockOrStatement (/Invend/apps/egbooking/node_modules/systemjs-builder/node_modules/traceur/bin/traceur.js:9217:32)
    at ParseTreeTransformer.transformIfStatement (/Invend/apps/egbooking/node_modules/systemjs-builder/node_modules/traceur/bin/traceur.js:9634:29)
    at IfStatement.$__super.transform (/Invend/apps/egbooking/node_modules/systemjs-builder/node_modules/traceur/bin/traceur.js:7438:28)
    at ParseTreeTransformer.transformAny (/Invend/apps/egbooking/node_modules/systemjs-builder/node_modules/traceur/bin/traceur.js:9193:44)
    at ParseTreeTransformer.transformList (/Invend/apps/egbooking/node_modules/systemjs-builder/node_modules/traceur/bin/traceur.js:9200:34)
    at ParseTreeTransformer.transformFunctionBody (/Invend/apps/egbooking/node_modules/systemjs-builder/node_modules/traceur/bin/traceur.js:9574:31)
[15:37:23] 'build' errored after 11 s
[15:37:23] Error in plugin 'run-sequence'
Message:
    An error occured in task 'jspm'.
```
Downgrading from `0.15.10.` to `0.15.9` fixed the problem.

**Comment by Lakerfield** [comment]
Fixed in aurelia. See: https://github.com/aurelia/loader-default/issues/31 and https://github.com/aurelia/bundler/issues/88

Run `jspm update` to fix

**Comment by guybedford** [comment]
Thanks, I've also added a fix in https://github.com/systemjs/builder/commit/facb4530e8cfd54bcdff5f815cd94e6406d2dbda to avoid the need for a workaround as well.

**Comment by BBosman** [comment]
I've updated my Aurelia and those errors are now fixed, but I still get the same error on other modules while bundling. Even with the facb453 commit applied to my local install.

```
[16:48:16] TypeError: Error compiling amd module "github:phiggins42/has.js@master/has.js" at jspm_packages/github/phiggins42/has.js@master/has.js
	Cannot read property 'parameters' of undefined
    at ParseTreeTransformer.AMDDefineRegisterTransformer.transformCallExpression (/Users/BBosman/Development/ExternalApps/Aurelia/node_modules/systemjs-builder/compilers/amd.js:229:76)
    at CallExpression.$__super.transform (/Users/BBosman/Development/ExternalApps/Aurelia/node_modules/traceur/bin/traceur.js:6649:28)
    at ParseTreeTransformer.transformAny (/Users/BBosman/Development/ExternalApps/Aurelia/node_modules/traceur/bin/traceur.js:9193:44)
    at ParseTreeTransformer.transformExpressionStatement (/Users/BBosman/Development/ExternalApps/Aurelia/node_modules/traceur/bin/traceur.js:9504:31)
    at ExpressionStatement.$__super.transform (/Users/BBosman/Development/ExternalApps/Aurelia/node_modules/traceur/bin/traceur.js:7120:28)
    at ParseTreeTransformer.transformAny (/Users/BBosman/Development/ExternalApps/Aurelia/node_modules/traceur/bin/traceur.js:9193:44)
    at ParseTreeTransformer.transformList (/Users/BBosman/Development/ExternalApps/Aurelia/node_modules/traceur/bin/traceur.js:9200:34)
    at ParseTreeTransformer.transformBlock (/Users/BBosman/Development/ExternalApps/Aurelia/node_modules/traceur/bin/traceur.js:9317:31)
    at Block.$__super.transform (/Users/BBosman/Development/ExternalApps/Aurelia/node_modules/traceur/bin/traceur.js:6612:28)
    at ParseTreeTransformer.transformAny (/Users/BBosman/Development/ExternalApps/Aurelia/node_modules/traceur/bin/traceur.js:9193:44)
    at ParseTreeTransformer.transformToBlockOrStatement (/Users/BBosman/Development/ExternalApps/Aurelia/node_modules/traceur/bin/traceur.js:9217:32)
    at ParseTreeTransformer.transformIfStatement (/Users/BBosman/Development/ExternalApps/Aurelia/node_modules/traceur/bin/traceur.js:9634:29)
    at IfStatement.$__super.transform (/Users/BBosman/Development/ExternalApps/Aurelia/node_modules/traceur/bin/traceur.js:7438:28)
    at ParseTreeTransformer.transformAny (/Users/BBosman/Development/ExternalApps/Aurelia/node_modules/traceur/bin/traceur.js:9193:44)
    at ParseTreeTransformer.transformList (/Users/BBosman/Development/ExternalApps/Aurelia/node_modules/traceur/bin/traceur.js:9200:34)
    at ParseTreeTransformer.transformFunctionBody (/Users/BBosman/Development/ExternalApps/Aurelia/node_modules/traceur/bin/traceur.js:9574:31)
    at FunctionBody.$__super.transform (/Users/BBosman/Development/ExternalApps/Aurelia/node_modules/traceur/bin/traceur.js:7293:28)
    at ParseTreeTransformer.transformAny (/Users/BBosman/Development/ExternalApps/Aurelia/node_modules/traceur/bin/traceur.js:9193:44)
    at ParseTreeTransformer.transformFunctionExpression (/Users/BBosman/Development/ExternalApps/Aurelia/node_modules/traceur/bin/traceur.js:9596:25)
    at ParseTreeTransformer.AMDDependenciesTransformer.transformFunctionExpression (/Users/BBosman/Development/ExternalApps/Aurelia/node_modules/systemjs-builder/compilers/amd.js:61:69)
    at FunctionExpression.$__super.transform (/Users/BBosman/Development/ExternalApps/Aurelia/node_modules/traceur/bin/traceur.js:7339:28)
    at ParseTreeTransformer.transformAny (/Users/BBosman/Development/ExternalApps/Aurelia/node_modules/traceur/bin/traceur.js:9193:44)
```

I tried if it was specific to that library, but when I temporarily removed `has` from my app I got the same error at another package:
```
[16:35:17] TypeError: Error compiling amd module "github:showdownjs/showdown@1.3.0/dist/showdown.js" at jspm_packages/github/showdownjs/showdown@1.3.0/dist/showdown.js
```

So the current fix doesn't seem to fix it.

**Comment by guybedford** [comment]
@BBosman thanks that was another bug, fixed in https://github.com/systemjs/builder/commit/f024acff22a90371e9b3222daccdad5a80e86da7.

**Comment by guybedford** [comment]
Released in 0.15.11.
